system/Helpers/url_helper.php

```diff
@@ -634,3 +634,51 @@ function mb_url_title(string $str, string $separator = '-', bool $lowercase = fa
 }
 
 //--------------------------------------------------------------------
+
+if (! function_exists('url_to'))
+{
+	/**
+	 * Get the full, absolute URL to a controller method
+	 * (with additional arguments)
+	 *
+	 * @param string $controller
+	 * @param mixed  ...$args
+	 *
+	 * @throws \CodeIgniter\Router\Exceptions\RouterException
+	 *
+	 * @return string
+	 */
+	function url_to(string $controller, ...$args): string
+	{
+		if (! $route = route_to($controller, ...$args))
+		{
+			if (strpos($controller, '::') !== false)
+			{
+				$class = explode('::', $controller)[0];
+
+				if (! class_exists($class))
+				{
+					$controller = service('routes')->getDefaultNamespace() . $controller;
+				}
+			}
+
+			if ($route = route_to($controller, ...$args))
+			{
+				return site_url($route);
+			}
```

Comment:
Almost there. These lines are unnecessary.

Reply:
@michalsn Sorry, but I don't quite get how they're unneccesary. What should it look like instead?

Reply:
We've already tried using `route_to()` on line 653 - so we know that will fail. Adding the default namespace will not change this.

Reply:
So now, I've got this:

```php
	function url_to(string $controller, ...$args): string
	{
		if (! $route = route_to($controller, ...$args))
		{
			$explode = explode('::', $controller);

			if (isset($explode[1]))
			{
				throw new \CodeIgniter\Router\Exceptions\RouterException(lang('HTTP.controllerNotFound', [$explode[0], $explode[1]]));
			}
			else
			{
				throw new \CodeIgniter\Router\Exceptions\RouterException(lang('HTTP.invalidRoute', [$controller]));
			}
		}

		return site_url($route);
	}
```

Works well with named routes and if you pass in a namespaced controller, but if I pass in a non-namespaced controller, it will throw an exception. That's why I added the default namespace. Am I still overlooking something?

Reply:
If we want to add the ability to omit the default namespace, this change should happen in the `route_to()` function. I just want to keep the consistency between functions. Of course, this is only my opinion.

Reply:
Alright, this makes perfect sense and consistency is important to me too. However, I'd like to know what others think about this (@MGatner @samsonasik @paulbalandan just to mention who's been involved in this thread). I'm perfectly fine with both having it in there and not having it in there.